middle_end/flambda2/to_cmm/to_cmm_set_of_closures.ml

```diff
@@ -473,6 +520,11 @@ let lift_set_of_closures env res ~body ~bound_vars layout set ~translate_expr
   then
     Misc.fatal_errorf "non-empty [updates] when lifting set of closures: %a"
       Set_of_closures.print set;
+  if not (Backend_var.Set.is_empty updates_free_names)
+  then
+    Misc.fatal_errorf
+      "non-empy free_names of [updates] when lifting set of closures: %a"
```

Comment:
`non-empy` -> `non-empty`
Besides, if we reach this point we have free names but no updates, so maybe the message could be more specific.